### PR TITLE
deps: use json-schema-ref-resolver

### PR DESCRIPTION
**Pull request by ivan-tymoshenko**
https://github.com/fastify/json-schema-ref-resolver

### CODE REVIEW

File: test/ref.test.js [hunk 1]
```diff
@@ -1377,7 +1377,7 @@ test('Bad key', t => {
       })
       t.fail('Should throw')
     } catch (err) {
-      t.equal(err.message, 'Cannot find reference "extrenal#/definitions/projectId"')
+      t.equal(err.message, 'Schema with id "extrenal" is not found.')
```
Comment (Eomm) [inline test/ref.test.js:1380]: the error message is less specific

**Review by Eomm — COMMENTED** [review]
I would mark this PR a breaking
I think we should battle-test the module before landing it - maybe test it in a real project

Ref: https://github.com/fastify/json-schema-ref-resolver/issues/1

File: test/ref.test.js (hunk 1 above)
Comment (ivan-tymoshenko) [reply test/ref.test.js:1380]: not a problem to fix that

**Comment by ivan-tymoshenko** [comment]
> I would mark this PR a breaking I think we should battle-test the module before landing it - maybe test it in a real project
> 
> Ref: [fastify/json-schema-ref-resolver#1](https://github.com/fastify/json-schema-ref-resolver/issues/1)

I don't see how it's a breacking change. Tests are passing. This package was developed from the [ref-resolver.js](https://github.com/fastify/fast-json-stringify/blob/master/lib/ref-resolver.js) module. If you look at implementation of the methods that are used, you will see that it's the same code.

File: test/ref.test.js (hunk 1 above)
Comment (ivan-tymoshenko) [reply test/ref.test.js:1380]: https://github.com/fastify/json-schema-ref-resolver/pull/4

**Review by Eomm — APPROVED** [review]
> The code that is used inside of fast-json-stringify wasn't changed.

Thks, did see it at first

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by mcollina** [comment]
@ivan-tymoshenko could you resolve the conflicts?

**Review by Uzlopak — APPROVED** [review]
LGTM

**Pull request merged**